packages/core/src/domain/session/sessionCookieStore.spec.ts

```diff
@@ -1,237 +1,38 @@
-          return cookie.currentValue()
-        }
-        const { currentState, retryState } = currentOnLockCheck()
-        cookie.setCurrentValue(buildSessionString(retryState))
-        return buildSessionString(currentState)
-      })
-    }
-
-    function buildSessionString(currentState: SessionState) {
-      return `${SESSION_COOKIE_NAME}=${toSessionString(currentState)}`
-    }
-
-    ;[
-      {
-        description: 'should wait for lock to be free',
-        lockConflict: 'onInitialLockCheck',
-      },
-      {
-        description: 'should retry if lock was acquired before process',
-        lockConflict: 'onAcquiredLockCheck',
-      },
-      {
-        description: 'should retry if lock was acquired after process',
-        lockConflict: 'onPostProcessLockCheck',
-      },
-      {
-        description: 'should retry if lock was acquired after persist',
-        lockConflict: 'onPostPersistLockCheck',
-      },
-    ].forEach(({ description, lockConflict }) => {
-      it(description, (done) => {
-        lockScenario({
-          [lockConflict]: () => ({
-            currentState: { ...initialSession, lock: 'locked' },
-            retryState: { ...initialSession, other: 'other' },
-          }),
-        })
-        persistSessionCookie(initialSession, COOKIE_OPTIONS)
-        processSpy.and.callFake((session) => ({ ...session, processed: 'processed' } as SessionState))
-
-        withCookieLockAccess({
-          options: COOKIE_OPTIONS,
-          process: processSpy,
-          after: (afterSession) => {
-            // session with 'other' value on process
-            expect(processSpy).toHaveBeenCalledWith({
-              ...initialSession,
-              other: 'other',
-              lock: jasmine.any(String),
-              expire: jasmine.any(String),
-            })
-
-            // end state with session 'other' and 'processed' value
-            const expectedSession = {
-              ...initialSession,
-              other: 'other',
-              processed: 'processed',
-              expire: jasmine.any(String),
-            }
-            expect(retrieveSessionCookie()).toEqual(expectedSession)
-            expect(afterSession).toEqual(expectedSession)
-            done()
-          },
-        })
-      })
-    })
-
-    it('should abort after a max number of retry', () => {
-      const clock = mockClock()
-
-      persistSessionCookie(initialSession, COOKIE_OPTIONS)
-      cookie.setSpy.calls.reset()
-
-      cookie.getSpy.and.returnValue(buildSessionString({ ...initialSession, lock: 'locked' }))
-      withCookieLockAccess({ options: COOKIE_OPTIONS, process: processSpy, after: afterSpy })
-
-      clock.tick(MAX_NUMBER_OF_LOCK_RETRIES * LOCK_RETRY_DELAY)
-      expect(processSpy).not.toHaveBeenCalled()
-      expect(afterSpy).not.toHaveBeenCalled()
-      expect(cookie.setSpy).not.toHaveBeenCalled()
-
-      clock.cleanup()
-    })
+  it('should persist a session in a cookie', () => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Should we test here that the session is actually in the cookie and not just what is returned by `retrieveSession`?

Reply:
👍